From 6bf4d00640caca476b77766c2ee6045ef7df95da Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:17 +0800
Subject: [PATCH] sw64: define NR_SYSCALLS as generated __NR_syscalls
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56XYC

--------------------------------

With this definition, the asm/unistd.h can keep unchanged if the
number of syscalls has changed.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/unistd.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/sw_64/include/asm/unistd.h b/arch/sw_64/include/asm/unistd.h
index b8854163d96f..6d1b8d1e2011 100644
--- a/arch/sw_64/include/asm/unistd.h
+++ b/arch/sw_64/include/asm/unistd.h
@@ -4,7 +4,7 @@
 
 #include <uapi/asm/unistd.h>
 
-#define NR_SYSCALLS			519
+#define NR_SYSCALLS			__NR_syscalls
 #define NR_syscalls			NR_SYSCALLS
 
 #define __ARCH_WANT_NEW_STAT
-- 
2.34.1

